### PR TITLE
WSGI

### PR DESCRIPTION
**Pull request by jtronge**
This is based on Rusty's original wsgi_final branch and should be able to run bee with wsgi. Right now the WFM is broken, but it should work once I merge in the workflow refactor code.

Some of the other changes that have been made are:

* Sqlite3 DB for the Scheduler and TM
* Use of sockets for REST calls
* Refactor of `beeflow`/`cli.py`
    * Use gunicorn for starting components
    * Handling of log output
    * Daemon support

### CODE REVIEW

**Comment by jtronge** [comment]
I think this should be working now. The `examples/cat-grep-tar` workflow runs and the integration test worked locally.

I ran into some issues with `examples/clamr-ffmpeg-build`. I think we might have had this problem before, but since it relies on a relative path (`beeflow/data/dockerfiles/Dockerfile.clamr-ffmpeg`), it fails when beeflow isn't running in the repo directory. I'm thinking it might be a good idea to require Dockerfiles to be in the workflow tarball.

**Comment by jtronge** [comment]
To launch everything, you can run `beeflow start`. By default this will launch a daemon process that then starts up all the components. This should still write a log to `~/.beeflow/logs/beeflow.log`. If you want to run it in the foreground you can launch with `beeflow start -F`.

You should also be able to kill everything with `beeflow stop` as well (that is everything except the gdb).

**Comment by pagrubel** [comment]
> I ran into some issues with `examples/clamr-ffmpeg-build`. I think we might have had this problem before, but since it relies on a relative path (`beeflow/data/dockerfiles/Dockerfile.clamr-ffmpeg`), it fails when beeflow isn't running in the repo directory. I'm thinking it might be a good idea to require Dockerfiles to be in the workflow tarball.

That is a good idea!

File: beeflow/tests/test_scheduler_rest.py [hunk 1]
```diff
@@ -92,7 +95,7 @@ def test_schedule_job_no_resources(scheduler):
     :param scheduler: url returned by scheduler fixture function
```
Comment (pagrubel) [inline beeflow/tests/test_scheduler_rest.py]: Did the param type change

**Review by pagrubel — COMMENTED** [review]
A few operational things I would like to see is a statement like when envoking beeflow start, just a statement like 
Starting Task Manager, Scheduler and Work Flow Manager.

I would also like to have a beeflow status showing what is running (could indicate health of the components, also could alert a user that they are on the wrong front end by showing nothing is running)

Finally I think there should be a warning from beeflow stop if workflows are running, atleast until we have some sort of resiliency built in.

File: beeflow/tests/test_scheduler_rest.py (hunk 1 above)
Comment (jtronge) [reply beeflow/tests/test_scheduler_rest.py]: Yes I think I had to change that. I'll need to fix that text there.

**Review by pagrubel — APPROVED** [review]

**Pull request merged**